src/app/tvshows/store/tvshows.actions.ts

```diff
@@ -0,0 +1,58 @@
+import { Action } from 'redux';
+import { TvShowData } from './tvshows.models';
+import { TvDbService } from '../../services';
+
+/**
+ * Actions key as enum types
+ *
+ * Typescripts's enum is ideal for this purpose, vs constants.  Plus
+ * we can use each enum value as a type specifier.  Combined with 'readonly' ensures
+ * actions keys are immutable and contain the intended value.
+ */
+
+export enum TypeKeys {
```

Comment:
The name of this enum is a bit vague. Consider maybe calling it `SearchActionTypes`.

Reply:
Done.